client/src/pages/TaskFormPage/TaskFormPage.js

```diff
@@ -1,40 +1,143 @@
-import React from "react";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
 import styles from "./TaskFormPage.module.css";
 import { InputBox } from "../../components/InputBox/InputBox";
 import { CredentialsButton } from "../../components/CredentialsButton/CredentialsButton.js";
+import { DateSelector } from "../../components/DateSelector/DateSelector.js";
+import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage.js";
+import { instance } from "../../axios";
 
 const TaskFormPage = () => {
+  const navigate = useNavigate();
+  const routeToCalendar = () => navigate("/calendar");
+
+  const [errorMessage, setErrorMessage] = useState("");
+  const [taskFormData, setTaskFormData] = useState({
+    title: "",
+    user_id: "620710de5b72a4271a59eabe", // TODO : figure out where to get it from
+    location: "",
+    description: "",
+    startDate: new Date(),
+    endDate: new Date(),
+    isStarted: false,
+  });
+
+  const updateTaskFormData = (e, value) => {
+    const newTaskFormData = { ...taskFormData };
+    newTaskFormData[value] =
+      value !== "startDate" && value !== "endDate" ? e.target.value : e;
+    setTaskFormData(newTaskFormData);
+    setErrorMessage("");
```

Comment:
We can probably remove this since we're only doing the validation check at the time the user clicks "submit/create task"

Reply:
This is to prevent a user from getting stuck with an error message. If we get a validation error and make some changes to our form, the error message shouldn't persist.